### ISSUE TITLE
[REQ] add prompt-completion dataset support

### ISSUE THREAD

**Issue by sozercan** [kaito-project/aikit#820]
Parent: #214

## Summary

Add `prompt-completion` as the second supported Unsloth SFT dataset type and introduce explicit runtime dispatch based on `datasets[].type`.

```yaml
datasets:
  - source: organization/question-answer-data
    type: prompt-completion
```

Expected record:

```json
{
  "prompt": "Question: What is a container image?\nAnswer:",
  "completion": " An immutable package containing an application and its dependencies."
}
```

Existing `type: alpaca` configurations must remain fully backward-compatible.

## Motivation

Go currently permits only `alpaca`, while `pkg/finetune/target_unsloth.py` ignores the configured type and always applies the Alpaca formatter. Adding another allowlisted value without Python dispatch would silently process the new schema as Alpaca.

Prompt-completion is the lowest-risk first extension because the frozen Unsloth/TRL stack already preserves `prompt` and `completion`, appends EOS when needed, and creates a completion mask without relying on a model-specific chat template or response-marker detection.

## Proposed changes

- Add a `prompt-completion` dataset type constant.
- Allow `alpaca` and `prompt-completion` in Go validation.
- Refactor the Python path into explicit dataset loading, schema validation, and preparation/dispatch steps.
- Make Python read `datasets[0].type` and reject unknown types rather than falling back to Alpaca.
- Preserve the current Alpaca path exactly:
  - Existing literal prompt.
  - Existing `instruction`, `input`, and `output` contract.
  - Explicit EOS append.
  - Full-sequence loss.
- For `prompt-completion`:
  - Require a non-empty dataset.
  - Require string `prompt` and non-empty string `completion` values.
  - Keep the two columns separate instead of converting them to `text`.
  - Use completion-only loss.
- Perform basic source/schema validation before model and LoRA allocation where possible.
- Add a JSONL fixture, aikitfile example, documentation, and a one-step GPU smoke path.

## Acceptance criteria

- Existing Alpaca aikitfiles require no changes.
- Alpaca rendering remains byte-for-byte equivalent and retains full-sequence loss.
- `type: prompt-completion` is accepted by Go and Python dispatch.
- Prompt-completion records never pass through `format_alpaca_examples`.
- Prompt tokens are masked and completion/EOS tokens are supervised.
- Missing, null, or invalid fields fail with a useful error before expensive model allocation where possible.
- Unknown types fail in both Go and Python.
- A one-step GPU smoke run trains and exports a non-empty GGUF.

## Tests

- Go validation tests for Alpaca, prompt-completion, and unknown types.
- Python tests for runtime dispatch, preserved Alpaca formatting, missing columns, invalid values, and failure ordering.
- Exact-lock preprocessing test proving completion-only labels.
- Add:
  - `test/fixtures/prompt-completion-smoke.jsonl`
  - `test/aikitfile-unsloth-prompt-completion-smoke.yaml`
- Run `make test`, `make test-python`, `make lint`, and the GPU fine-tune/export smoke test.

## Out of scope

- Messages or ShareGPT.
- Raw text.
- Column mappings.
- Alternate splits/subsets/revisions/checksums.
- Multiple datasets.
- Strict YAML behavior changes.
- Response-only chat loss.
- DPO, ORPO, KTO, GRPO, or multimodal objectives.
